docs/DeveloperGuide.md

```diff
@@ -1,34 +1,413 @@
+        string into its individual components.
+        
+Step 4. Each subsequent string separated by a space is stored in a string array named `commandArr`. The 0th index of the 
+        `commandArr` array is the first word, the 1st index is the second word, and so on. In this case there is only 
+        one word stored in the array, at the 0th index, which is “help”.
+        
+Step 5. The string at the 0th index is then used in a switch statement, where each case represents the different menu 
+        options available. As such, the contents of the case with reference “help” is run, which is a return statement 
+        containing a new `HelpCommand()`. This leads to the execution of the `help` command.
+        
+#### 2.3.3. Design of QuizQuestionsManager
+
+To start a quiz in E-Duke-8, the user will have to indicate the number of questions that he wants to attempt, as well as the topic to get the questions from. Thereafter, questions will be shown to the user one by one until all them are attempted. 
+
+The class diagram given below explains the high-level design of the Quiz system in E-Duke-8. Given below it is a quick overview of each component.
+
+![QuizQuestionsManager_Class_Diagram](./images/QuizQuestionsManager.png)
+
+An object of `SingleTopicQuiz` class represents an instance of the quiz in E-Duke-8. Its `numberOfQuestions` attribute and `Topic` object correspond to the user's specified number of questions and topic for the quiz respectively.
+
+The `startQuiz(:Ui)` method call from the `SingleTopicQuiz` object initializes an object of `QuizQuestionsManager` by passing into it `numberOfQuestions`, as well as an ArrayList of questions from the `Topic` object. The `QuizQuestionsManager` object will then randomly select `numberOfQuestions` questions from the topic the user has chosen, using its `setQuizQuestions(:int, :ArrayList<Displayable>)` method. 
+
+Thereafter, by making use of `QuizQuestionsManager`'s `getNextQuestion()` and `areAllQuestionsAnswered()` method calls, the `goThroughQuizQuestions(:Ui, :QuizQuestionsManager)` will loop through the questions until the user has answered all of them on the command line interface.
+
+#### 2.3.4. Implementation of QuizQuestionsManager
+
+As mentioned earlier in the section on the design of the quiz system, a `QuizQuestionsManager` object will randomly select the indicated number of questions from the list of questions in the `Topic` object, and these will form the quiz questions for the user.
+
+The sequence diagram below shows how `QuizQuestionsManager` is implemented to achieve this for the scenario where the user indicates that he wants to attempt 5 questions from the topic on OOP, which translates to the `setQuizQuestions(5, questionsInTopic)` call:
+
+
+![QuizQuestionsManager::setQuizQuestions_Sequence_Diagram](./images/QuizQuestionsManager_setQuizQuestions.png)
```

Comment:
![image](https://user-images.githubusercontent.com/56167146/97656514-9ca4c400-1aa2-11eb-8cef-43f86970276d.png)
The ending for :ArrayList<Displayable> could perhaps end with the return arrow.

Reply:
Thank you for spotting that! I have fixed it and will update the diagram soon!